great_expectations/rule_based_profiler/config/base.py

```diff
@@ -303,6 +303,29 @@ def get_config_class(cls) -> Type["RuleBasedProfilerConfig"]:  # noqa: F821
     def get_schema_class(cls) -> Type["RuleBasedProfilerConfigSchema"]:  # noqa: F821
         return RuleBasedProfilerConfigSchema
 
+    # noinspection PyUnusedLocal,PyUnresolvedReferences
+    @classmethod
+    def resolve_config_using_acceptable_arguments(
+        cls,
+        profiler: "RuleBasedProfiler",  # noqa: F821
+        variables: Optional[Dict[str, Any]] = None,
+        rules: Optional[Dict[str, Dict[str, Any]]] = None,
+    ) -> "RuleBasedProfilerConfig":
+        runtime_config: RuleBasedProfilerConfig = profiler.config
+
+        # If applicable, override config attributes with runtime args
+        runtime_rules: dict = rules if rules is not None else runtime_config.rules
+        runtime_variables: Optional[dict] = (
+            variables if variables is not None else runtime_config.variables
+        )
+
+        return cls(
```

Comment:
@cdkini I may be missing the crux of the matter here -- I was hoping to see something like one of our reconciliation methods run taking in the existing configuration (or parts thereof) and the supplied overrides (e.g., like we have in the tests of the reconciliation logic as well as, most recently, utilized for self-initializing expectations).  Happy to discuss.

Reply:
UPDATE - per our discussion, I've leveraged the existing logic to aid with reconciliation. I've also reconciled the variables since it makes sense to keep the logic the same between usage stats and the actual run method.